code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/EnsureSettingForProvider.java

```diff
@@ -0,0 +1,73 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.api.auth.Identity;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.SettingsUtils;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubUtils;
+import io.cattle.platform.util.type.InitializationTask;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class EnsureSettingForProvider implements InitializationTask {
+
+    private static final Log logger = LogFactory.getLog(EnsureSettingForProvider.class);
+
+    @Inject
+    SettingsUtils settingsUtils;
+
+    @Inject
+    GithubUtils githubUtils;
+
+    @Override
+    public void start(){
+        if (SecurityConstants.SECURITY.get() &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_SECRET.get()) &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_ID.get())){
+            if (StringUtils.isBlank(SecurityConstants.AUTH_PROVIDER.get())) {
+                settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, GithubConstants.CONFIG);
+                logger.info("Upgrade detected. Adding setting for :" + SecurityConstants
+                        .AUTH_PROVIDER_SETTING);
+
+            }
+            if ("restricted".equalsIgnoreCase(GithubConstants.ACCESS_MODE.get()) &&
+                    StringUtils.isBlank(GithubConstants.ALLOWED_IDENTITIES_SETTING)){
+                List<Identity> allowedIdentities = new ArrayList<>();
+                String allowedUsers = ArchaiusUtil.getString("api.auth.github.allowed.users").get();
+                List<String> userIds = githubUtils.fromCommaSeparatedString(allowedUsers);
+                for (String userId: userIds){
+                    Identity identity = new Identity(GithubConstants.USER_SCOPE, userId.split(":")[1]);
+                    allowedIdentities.add(identity);
+                }
+                String allowedOrgs = ArchaiusUtil.getString("api.auth.github.allowed.orgs").get();
+                List<String> orgIds = githubUtils.fromCommaSeparatedString(allowedOrgs);
+                for (String orgId: orgIds){
+                    Identity identity = new Identity(GithubConstants.ORG_SCOPE, orgId.split(":")[1]);
+                    allowedIdentities.add(identity);
+                }
+                Iterator<Identity> identityIterator = allowedIdentities.iterator();
+                StringBuilder sb= new StringBuilder();
+                while (identityIterator.hasNext()){
+                    sb.append(identityIterator.next().getId().trim());
+                    if (identityIterator.hasNext()) sb.append(',');
```

Comment:
Can you please not do this one line if statement? Its a very unnatural syntax for java